### DISCUSSION TITLE
Another question for how to contribute to this project

### DISCUSSION THREAD

**Discussion by haphamdev** [fwcd/kotlin-language-server#402] [Q&A] [answered]
I am a big fan of Kotlin and language server. 
I would like to contribute to this project as a newbie.
I know Kotlin but I don't know much about how a language server is implemented.
Where should I start? Is there a list of tasks which I can pick?

**Comment by fwcd** [comment, marked as answer]
Hi, to get started developing, you want to be able to build the language server locally on your machine. If you're using VSCode, the easiest way to get started is to use the [Kotlin extension](https://marketplace.visualstudio.com/items?itemName=fwcd.kotlin). Make sure to have Java installed, then run

```sh
./gradlew :server:installDist
```

to compile and package the language server (see [`BUILDING.md`](https://github.com/fwcd/kotlin-language-server/blob/main/BUILDING.md) for details). Then, by adding the following setting to your VSCode configuration:

```json
{
  "kotlin.languageServer.path": "<path to cloned kotlin-language-server repo>/server/build/install/server/bin/kotlin-language-server"
}
```

and reloading your VSCode instance you should be able to use your newly built language server in a Kotlin project of your choice (e.g. `kotlin-language-server` itself). You can check out the [list of issues](https://github.com/fwcd/kotlin-language-server/issues), specifically [those labeled `good first issue`](https://github.com/fwcd/kotlin-language-server/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)) for some ideas what features could be worked on, but of course feel free to come up with your own ideas! Bug reports or fixes are of course always welcome too.

If you have any questions, there's [a Discord server](https://discord.gg/cNtppzN) for chatting.

Cheers!
